packages/api-request/src/middlewares/test/preloading.js

```diff
@@ -0,0 +1,41 @@
+import createPreloadingMiddleware from '../preloading';
+
+describe( 'Preloading Middleware', () => {
+	it( 'should return the preloaded data if provided', () => {
+		const body = {
+			status: 'this is the preloaded response',
+		};
+		const preloadedData = {
+			'wp/v2/posts': {
+				body,
+			},
+		};
+		const prelooadingMiddleware = createPreloadingMiddleware( preloadedData );
+		const requestOptions = {
+			method: 'GET',
+			path: 'wp/v2/posts',
+		};
+
+		const response = prelooadingMiddleware( requestOptions );
+		response.then( ( value ) => {
```

Comment:
I think returning this statement will ensure that Promise got completed.

Reply:
This is not a regular promise (jQuery), so I went with the `done` trick